### ISSUE TITLE
[Info] Packaging for Fedora (rpm)

### ISSUE THREAD

**Issue by dreua** [cnlohr/cnping#49]
I built a cnping package for Fedora, check it out here: https://copr.fedorainfracloud.org/coprs/dreua/cnping/
In case someone is interested, here is the specfile: 
https://copr-be.cloud.fedoraproject.org/results/dreua/cnping/fedora-28-x86_64/00779658-cnping/cnping.spec

**Comment by cnlohr** [comment]
Does that mean it may become part of the distro?  I.e. available by yum install?

**Comment by dreua** [comment]
Yes, it *may* become part of the main repository eventually, that depends on whether I find a sponsor and manage to get this reviewed. But I have to learn more about the packaging process and fix some minor issues first.
Copr for Fedora is like AUR on Arch or PPA on Ubuntu/Debian, you can already easily install the package by: `dnf copr enable dreua/cnping` to enable the copr as a package source and `dnf install cnping`. (dnf is the new yum.) It will also update like a real package. 
When enabling the copr you will of course get a warning that this "is not part of the main distribution, and quality may vary."

**Comment by cnlohr** [comment]
Understood.  Hmm, I would love to see what it would take to make this part of an actual distro.  Who knows, maybe someday even Debian!

**Comment by dreua** [comment]
You mean you'd like to see a todo list? Here you go:

- [ ] Include the whole license text(s) in the license file
- [ ] Write and include a man page
- [ ] Make releases. (Simple git tags would be fine. The versioning scheme is important for the package manager to figure out which of two packages is older/newer. Using the date and git hashes is possible but somewhat messy imo.)
- [ ] Release source bundles without binaries. Github already creates source bundles for each commit and tag for us but [it should not include binaries](https://fedoraproject.org/wiki/Packaging:Guidelines?rd=Packaging/Guidelines#No_inclusion_of_pre-built_binaries_or_libraries), i.e. cnping.exe. The exe might find it's new home on the [release page.](https://github.com/cnlohr/cnping/releases) It could be automatically built and uploaded there by Travis-CI, see #32 

Todos especially for the packager (me):
- [ ] Rebuild latest version with latest spec.
- [ ] Check that the build system's compiler flags are respected.
- [ ] Build and test builds with rpmlint and fedora-review
- [ ] Introduce in the Fedora-Dev mailinglist.
- [ ] Create and monitor review request bug
- [ ] Help reviewing other requests
- [ ] Get sponsored
- [ ] To be continued...

**Comment by dreua** [comment]
No, actually I just wrote down the todos to get them out of my head and did't think about assigning them to anyone. I edited the list to what seems reasonable to me. Maybe I can do PRs for the first two in the next days but you can also do it yourself of course ;) 
About IPv6: We can always update the package later when it's ready, so there is no real need to do this now.
